src/pages/TabDetails.tsx

```diff
@@ -161,7 +161,8 @@ const ListProgress = () => {
 
   const list = cycles
     .filter((_value, idx) => {
-      return idx > 0;
+      // NOTE: 6 is the number of cycles we display in details. We store a maximum of 7 cycles (in case the last cycle is accidentally deleted)
+      return idx > 0 && idx < 6;
```

Comment:
Why did you use `filter` instead of `slice`?

Reply:
Fixed